src/lib.rs

```diff
@@ -437,3 +438,73 @@ pub mod prelude {
 
     pub use crate::read_only::{ReadOnlyOption, ReadState};
 }
+
+/// Build a logger for tests.
+///
+/// Currently, this is a terminal log. It ensures it is only initialized once to prevent clobbering.
+// This is `pub` so that testing and benching functions can use it.
+// `#[cfg(test)]` doesn't work for benching.
+//
+// TODO: Investigate reporting strategies. Tests are run multithreaded in general, and we'd like to be able to sort through and understand test results.
+#[doc(hidden)]
+pub fn testing_logger() -> &'static Logger {
+    use std::sync::Once;
+    static LOGGER_INITIALIZED: Once = Once::new();
+    static mut LOGGER: Option<Logger> = None;
+
+    unsafe {
+        LOGGER_INITIALIZED.call_once(|| {
+            let decorator = slog_term::TermDecorator::new().build();
+            let drain = slog_term::CompactFormat::new(decorator).build().fuse();
+            let drain = slog_envlogger::new(drain).fuse();
+            let drain = slog_async::Async::new(drain)
```

Comment:
For testing having a sync logger might be preferred, this way the stdout doesn't get mangled when interwoven with the testing output.


```
test integration_cases::test_raft::test_recv_msg_request_vote ...  ok
idtest integration_cases::test_raft::test_restore_with_learner ... ok:
test integration_cases::test_raft::test_send_append_for_progress_probe ...  ok
3test integration_cases::test_raft::test_send_append_for_progress_replicate ... ok

test integration_cases::test_raft::test_send_append_for_progress_snapshot ...  ok
test integration_cases::test_raft::test_single_node_candidate ... Febok
25test integration_cases::test_raft::test_single_node_commit ...  thread 'ok10integration_cases::test_raft::test_state_transition
:' panicked at 'test integration_cases::test_raft::test_single_node_pre_candidate ... 37assertion failed: `(left != right)`
  left: `Follower`,
 right: `Follower`: invalid transition [follower -> leader]ok:',
14src/raft.rstest integration_cases::test_raft::test_step_config ... .:ok166942
:test integration_cases::test_raft::test_step_ignore_config ...  9ok

ERROnote: Run with `RUST_BACKTRACE=1` for a backtrace.
test integration_cases::test_raft::test_transfer_non_member ... ok
test integration_cases::test_raft::test_vote_from_any_state ...  can't become learner when restores snapshot [index: 11, term: 11]okthread '
integration_cases::test_raft::test_state_transition, ' panicked at 'assertion failed: `(left != right)`
  left: `Leader`,
 right: `Leader`: invalid transition [leader -> pre-candidate]snapshot_term', src/raft.rs::913 :911
, snapshot_index: 11, tag:
thread 'integration_cases::test_raft::test_state_transition' panicked at 'assertion failed: `(left != right)`
  left: `Leader`,
 right: `Leader`: invalid transition [leader -> candidate]', src/raft.rs:889:9
```